src/DataFetcher.py

```diff
@@ -8,322 +9,328 @@
+        if not self.stockrow_key_stats:
+            self.lock.release()
+            return
+        success = self.stockrow_key_stats.parse_json_data(response.content)
+        if not success:
+            self.stockrow_key_stats = None
+        self.lock.release()
+
+    def fetch_pe_ratios(self):
+        """
+        Fetching PE Ratios to calculate Sticker Price and Safety Margin Price
+        First we need to get an internal MSN stock id for a ticker
+        and then fetch PE Ratios.
+        """
+        self.pe_ratios = MSNMoney(self.ticker_symbol)
+        session = self._create_session()
+        rpc = session.get(self.pe_ratios.get_ticker_autocomplete_url(), allow_redirects=True, hooks={
+            'response': self.continue_fetching_pe_ratios,
+        })
+        self.rpcs.append(rpc)
+
+    def continue_fetching_pe_ratios(self, response, *args, **kwargs):
+        """
+        After msn_stock_id was fetched in fetch_pe_ratios method
+        we can now get the financials
+        """
+        msn_stock_id = self.pe_ratios.extract_stock_id(response.text)
+        session = self._create_session()
+        rpc = session.get(self.pe_ratios.get_key_ratios_url(msn_stock_id), allow_redirects=True, hooks={
+            'response': self.parse_pe_ratios,
+        })
+        self.rpcs.append(rpc)
+
+    # Called asynchronously upon completion of the URL fetch from
+    # `fetch_pe_ratios` and `continue_fetching_pe_ratios`.
+    def parse_pe_ratios(self, response, *args, **kwargs):
+        if response.status_code != 200:
+            return
+        if not self.pe_ratios:
+            return
+        result = response.text
+        success = self.pe_ratios.parse_pe_ratios(result)
+        if not success:
+            self.pe_ratios = None
+
+    def fetch_yahoo_finance_analysis(self):
+        self.yahoo_finance_analysis = YahooFinanceAnalysis(self.ticker_symbol)
+        session = self._create_session()
+        rpc = session.get(self.yahoo_finance_analysis.url, allow_redirects=True, hooks={
+            'response': self.parse_yahoo_finance_analysis,
+        })
+        self.rpcs.append(rpc)
+
+    # Called asynchronously upon completion of the URL fetch from
+    # `fetch_yahoo_finance_analysis`.
+    def parse_yahoo_finance_analysis(self, response, *args, **kwargs):
+        if response.status_code != 200:
+            return
+        if not self.yahoo_finance_analysis:
+            return
+        result = response.text
+        success = self.yahoo_finance_analysis.parse_analyst_five_year_growth_rate(result)
+        if not success:
+            self.yahoo_finance_analysis = None
+
+    def fetch_yahoo_finance_quote(self):
+        self.yahoo_finance_quote = YahooFinanceQuote(self.ticker_symbol)
+        session = self._create_session()
+        rpc = session.get(self.yahoo_finance_quote.url, allow_redirects=True, hooks={
+            'response': self.parse_yahoo_finance_quote,
+        })
+        self.rpcs.append(rpc)
```

Comment:
Looks like all the lines from 134-235 are also just whitespace that can be reverted

Reply:
If you think that some changes may be reverted, no problem, sorry again

Reply:
I left some comments to try to make it easier to highlight what to change. Basically it would be great to only have the actual modified lines changed in the final PR. It makes maintaining the `blame` history easier.

Reply:
Well, it's true